### PR TITLE
Fixing Get Upload By Id Arm Template Bug

### PR DESCRIPTION
**Pull request by ccaprio97**
## What’s changing?

updating arm template to have each states app id

## Why?

NAC-1411

## This PR has:

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

**Comment by ccaprio97** [comment]
forgot to remove all the echo statements and put the workplace id back into create apim bash. will do

File: iac/create-apim.bash [hunk 1]
```diff
@@ -118,6 +118,9 @@ main () {
       --filter "displayName eq '${orch_name}'" \
       --query "[0].appId" \
       --output tsv)
+  echo "HERE IS THE ORCH ID!!!!!!"
```
Comment (bgfoshay) [inline iac/create-apim.bash:121]: Nitpick: Did you mean to leave these echo statements in here?

File: iac/create-apim.bash [hunk 2]
```diff
@@ -138,6 +141,39 @@ main () {
...
+
+  local state_app_id_list=()
+
+  ea_etl_app_id=$(\
+    az ad app list \
+      --display-name "cc-func-eaetl-cjc" \
+      --filter "displayName eq 'cc-func-eaetl-cjc'" \
+      --query "[0].appId" \
+      --output tsv)
+  #echo "HERE IS THE EA ETL APP ID!!!!!!!!!!!!!!!!!!!!!!!!!"
```
Comment (bgfoshay) [inline iac/create-apim.bash:154]: Nitpick: commented out debugging code?

File: iac/create-apim.bash [hunk 3]
```diff
@@ -164,7 +200,6 @@ main () {
         diagnosticSettingName="${DIAGNOSTIC_SETTINGS_NAME}" \
         eventHubAuthorizationRuleId="${EH_RULE_ID}" \
         eventHubName="${EVENT_HUB_NAME}" \
-        workspaceId="${LOG_ANALYTICS_WORKSPACE_ID}" \
```
Comment (bgfoshay) [inline iac/create-apim.bash:167]: Question: Was the removal of this parameter intentional?

**Review by bgfoshay — CHANGES_REQUESTED** [review]
The approach looks good. The comments and debug statements could use some cleaning up. And there's one section that's hardcoded to your resources that needs to be fixed. But overall, good job figuring out a way to do the replacements.

File: iac/arm-templates/apim-bulkupload.json [hunk 4]
```diff
@@ -28,6 +28,9 @@
         },
         "allUploadOperationsPolicyXml": {
             "type": "String"
+        },
```
Comment (ryangeorgeharvey) [inline iac/arm-templates/apim-bulkupload.json:31]: nitpick (non-blocking): Add a description for the ARM variable.

File: iac/create-apim.bash [hunk 5]
```diff
@@ -139,6 +139,18 @@ main () {
...
+  local state_app_id_list=()
+  for i in $state_abbrs
+    do
+    single_state_etl_app_id=$(\
+    az ad app list \
+      --display-name "${PREFIX}-func-${i}etl-${ENV}" \
+      --filter "displayName eq '${PREFIX}-func-${i}etl-${ENV}'" \
+      --query "[0].appId" \
+      --output tsv)
+      state_app_id_list+=("${single_state_etl_app_id}")
```
Comment (ryangeorgeharvey) [inline iac/create-apim.bash:151]: nitpick (non-blocking): line 151 is indented too much. It should be aligned with line 145.

**Review by ryangeorgeharvey — APPROVED** [review]
Changes LGTM, but recommend making the minor fixes.

**Review by bgfoshay — APPROVED** [review]
👍

**Pull request merged**